uikit/viewmodels/src/main/java/com/sygic/maps/uikit/viewmodels/searchresultlist/SearchResultListViewModel.kt

```diff
@@ -0,0 +1,102 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.uikit.viewmodels.searchresultlist
+
+import android.view.View
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.recyclerview.widget.RecyclerView
+import com.sygic.maps.tools.annotations.AutoFactory
+import com.sygic.maps.uikit.viewmodels.common.extensions.toSearchResultList
+import com.sygic.maps.uikit.viewmodels.common.search.SearchManager
+import com.sygic.maps.uikit.views.common.extensions.asSingleEvent
+import com.sygic.maps.uikit.views.common.extensions.hideKeyboard
+import com.sygic.maps.uikit.views.common.livedata.SingleLiveEvent
+import com.sygic.maps.uikit.views.searchresultlist.SearchResultList
+import com.sygic.maps.uikit.views.searchresultlist.adapter.DefaultStateAdapter
+import com.sygic.maps.uikit.views.searchresultlist.adapter.ResultListAdapter
+import com.sygic.maps.uikit.views.searchresultlist.adapter.SearchResultListAdapter
+import com.sygic.maps.uikit.views.searchresultlist.data.SearchResultItem
+import com.sygic.sdk.search.Search
+import com.sygic.sdk.search.SearchResult
+
+/**
+ * A [SearchResultListViewModel] is a basic ViewModel implementation for the [SearchResultList] class. It listens to
+ * the Sygic SDK [Search.SearchResultsListener] and updates the search result list in the [SearchResultList] view.
+ */
+@AutoFactory
+@Suppress("unused", "MemberVisibilityCanBePrivate")
+open class SearchResultListViewModel internal constructor(
+    private val searchManager: SearchManager
+) : ViewModel(), DefaultLifecycleObserver, ResultListAdapter.ClickListener<SearchResult> {
+
+    val onSearchResultItemClickObservable: LiveData<SearchResultItem<out SearchResult>> = SingleLiveEvent()
+    val searchResultListDataChangedObservable: LiveData<List<SearchResultItem<out SearchResult>>> = SingleLiveEvent()
+
+    val activeAdapter: MutableLiveData<ResultListAdapter<SearchResult, ResultListAdapter.ItemViewHolder<SearchResult>>> = MutableLiveData()
+
+    private val defaultStateAdapter = DefaultStateAdapter<SearchResult>()
+    private val resultListAdapter = SearchResultListAdapter<SearchResult>()
+
+    private var lastScrollState = RecyclerView.SCROLL_STATE_IDLE
+
+    private val searchResultsListener = Search.SearchResultsListener { input, _, results ->
+        results.toSearchResultList().let {
+            resultListAdapter.items = it
```

Comment:
tu sa vysledky z sdk daju do adapteru, tj UI sa updatne, jo? a tie druhe live data potom updatuju co?

Reply:
ano, items mají overridnutý setter a interně si volají notifyDataSetChanged(). to druhé je jak si možná vzpomínáš kvůli tomu multiresult kliku z klávesnice, který je v SearchFragmentViewModel-u. nakonec jsme se dohodli na tomto řešení, pokud si dobře vzpomínám.